internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -242,14 +242,23 @@ func (r *Reconciler) reconcileVariables(ctx context.Context, clusterClass *clust
 				continue
 			}
 			if resp.Variables != nil {
+				uniqueNamesForPatch := map[string]bool{}
```

Comment:
nit (feel free to ignore)
```suggestion
				variableNames := sets.Set[string]()
```
so below you can uset variableNames.Has and variableNames.Insert which reads better